File: 백준/Silver/1246. 온라인 판매/온라인 판매.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        int [] prices = new int[m];
        for(int i = 0;i<m;i++){
            prices[i] = Integer.parseInt(br.readLine());
        }
        Arrays.sort(prices);

        int maxRevenue = 0;
        int bestPrices = 0;

        for(int i = 0;i<m;i++){
            int price = prices[i];
            int buyers = Math.min(n,m-i);
            int revenue = price* buyers;

            if(revenue>maxRevenue){
                maxRevenue = revenue;
                bestPrices = price;
            }
        }
        System.out.println(bestPrices+" "+ maxRevenue);
    }
}

File: 백준/Silver/1246. 온라인 판매/온라인 판매.py
import sys
input = sys.stdin.readline

n,m = map(int,input().split())
pi = list(int(input()) for _ in range(m))
pi.sort()

max_revenue = 0
best_price = 0
for i in range(m):
    price = pi[i]
    buyers = min(n,m-i)
    revenue = price*buyers

    if revenue>max_revenue:
        max_revenue = revenue
        best_price = price
print(best_price, max_revenue)

